src/App.tsx

```diff
@@ -1,56 +1,108 @@
-import React from 'react';
+import cn from 'classnames';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+type Person = {
+  name: string,
+  sex: string,
+  born: number,
+  died: number,
+  fatherName: string | null,
+  motherName: string | null,
+  slug: string,
+};
+
+type Props = {
+  delay: number,
+};
+
+export const App: React.FC<Props> = ({ delay }) => {
+  const [query, setQuery] = useState('');
+
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const [isFocused, setIsFocused] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
+    [],
+  );
```

Comment:
Need to fix it
<img width="1169" alt="image" src="https://github.com/mate-academy/react_autocomplete/assets/64750958/de25d3bb-1d27-4560-b1d0-bdbd97b0b08f">